### ISSUE TITLE
UI.navigate() + BeforeLeaveEvent.postpone() updates the url before proceed().

### ISSUE THREAD

**Issue by bogdanudrescu** [vaadin/flow#8122]
Doing UI.navigate() and then postpone() should not update the URL but only after the navigation is continued.

Original issue https://github.com/vaadin/flow/issues/3619

**Comment by bogdanudrescu** [comment]
Closing due to being fixed together with the RouterLink fix by https://github.com/vaadin/flow/pull/8125
